src/server/db_slice.cc

```diff
@@ -555,66 +566,24 @@ tuple<PrimeIterator, ExpireIterator, bool> DbSlice::AddOrFind2(const Context& cn
     // evicted_obj_bytes = EvictObjects(-evp.mem_budget(), it, &db);
   }
 
-  if (inserted) {  // new entry
-    db.stats.inline_keys += it->first.IsInline();
-    int64_t delta = it->first.MallocUsed();
-    db.stats.obj_memory_usage += delta;
-    db.stats.AddTypeMemoryUsage(it->second.ObjType(), delta);
+  db.stats.inline_keys += it->first.IsInline();
+  AccountObjectMemory(it->first, &db.stats, 1);  // Account for key
 
-    events_.garbage_collected = db.prime.garbage_collected();
-    events_.stash_unloaded = db.prime.stash_unloaded();
-    events_.evicted_keys += evp.evicted();
-    events_.garbage_checked += evp.checked();
+  DCHECK_EQ(it->second.MallocUsed(), 0UL);  // Make sure accounting is no-op
+  PreUpdate(cntx.db_index, it, PreUpdateMode::kWithoutCallbacks);
 
```

Comment:
I dont think you should call PreUpdate here as PreUpdate should be call before we update the value of an entry. The only common functionality here is the SetVerstion and you can call it explicitly

Reply:
I find this more future compatible, but any changes we may make in `PreUpdate()`.
But since you prefer that way, I won't argue. Done. I'd be happy to revert if you change your mind though :)

Reply:
note: you dont need the PreUpdateMode enum now.